server/aesdsocket.c

```diff
@@ -0,0 +1,346 @@
+  char recPacket[BUFFER_LENGTH];
+  char* recBuffer = NULL;
+  char* trsBuffer = NULL;
+  
+  /******************** ACCEPT CONNECTION TILL SIGNAL CAUGHT *******************/
+  while(!signalCaught){                                                                            // Execute till SIGTERM or SIGINT //
+    char ipString[INET6_ADDRSTRLEN];
+    clientFd = accept(sockFd, (struct sockaddr*)&peerAddr, &peerAddrLen);                          // Accept connection //
+    if(clientFd == STATUS_FAILURE){
+      syslog(LOG_ERR, "Cannot accept connection \r\n");
+      return EXIT_FAILURE;
+    }
+    else{
+      inet_ntop(peerAddr.sa_family, getAddress((struct sockaddr*)&peerAddr), ipString, sizeof(ipString));
+      syslog(LOG_INFO, "Connection accepted - %s \r\n", ipString);                                 // Print IP address //
+      clientFdOpenFlag = true;
+    }
+    
+    recBuffer = (char*) malloc(BUFFER_LENGTH*sizeof(char)*numBufFragments);                        // Allocate overall buffer //
+    if(recBuffer == NULL){
+      syslog(LOG_ERR, "Malloc failed \r\n");
+      return EXIT_FAILURE;
+    }
+
+    while((bytesRcvd = recv(clientFd, recPacket, BUFFER_LENGTH, 0)) > 0){                          // Receive data over socket till it is available //
+      if((numBufFragments*BUFFER_LENGTH) - noOfBytesinBuffer < bytesRcvd){                         // Check if the buffer size has to be increased //
+        recBuffer = (char*)realloc(recBuffer, (++numBufFragments * BUFFER_LENGTH));                // Realloc buffer //
+        if(recBuffer == NULL){
+          syslog(LOG_ERR, "realloc failed \r\n");
+          return EXIT_FAILURE;
+        }
+      }
+      memcpy(noOfBytesinBuffer+recBuffer, recPacket, bytesRcvd);                                   // Copy data to the overall buffer //
+      noOfBytesinBuffer += bytesRcvd;
+      bool newLine = false;
+      int ind;
+      for (ind=0; ind<noOfBytesinBuffer; ind++){
+        if (recBuffer[ind] == '\n'){                                                               // Check for new Line input //
+          newLine = true;
+          break;
+        }
+      }
+      if (newLine){            
+        int bytesChanged = ind + 1;
+        bytesPerPacket = write(fileFd_W_R, recBuffer, bytesChanged);                               // Write the data to file Descriptor //
+        if(bytesPerPacket == STATUS_FAILURE){
+          syslog(LOG_ERR, "File write system call failed \r\n");
+          return EXIT_FAILURE;
+        }
+        else if(bytesPerPacket < bytesChanged){
+          syslog(LOG_ERR, "File write is incomplete \r\n");
+          return EXIT_FAILURE;
+        }
+        else{
+          bytesToBeWritten += bytesPerPacket;
+          noOfBytesinBuffer -= (ind + 1);
+          bytesChanged = bytesToBeWritten;
+          trsBuffer = (char*)malloc(bytesChanged);                                                 // Allocate memory to store the read data //
```

Comment:
@Vignesh-Vadivel what assumption does this make regarding the size of the process heap and the total size of all packets sent?